lib/mbaasRequest/mbaasRequest.js

```diff
@@ -175,6 +175,15 @@ function responseHandler(params, cb) {
 
     if (err || (httpResponse && httpResponse.statusCode >= 400)) {
 
+      // MbaaS is unreachable
+      if (err.code !== undefined && err.code === "ENOTFOUND" || err.code === "ETIMEDOUT") {
```

Comment:
I think the idea is to have a relevant error message for all errors returned from an MBaaS so I don't think

`err.code === "ENOTFOUND" || err.code === "ETIMEDOUT"` is required.